provider/aws/registries.go

```diff
@@ -52,13 +52,15 @@ func (p *Provider) RegistryAdd(server, username, password string) (*structs.Regi
 			return nil, fmt.Errorf("unable to authenticate")
 		}
 	default:
-		_, err := dc.AuthCheck(&docker.AuthConfiguration{
-			ServerAddress: server,
-			Username:      username,
-			Password:      password,
-		})
-		if err != nil {
-			return nil, fmt.Errorf("unable to authenticate")
+		if !(p.PrivateBuild) {		
```

Comment:
This should probably just be the first case of the "validate login" switch with no body